lib/model/CEventRateBucketGatherer.cc

```diff
@@ -223,23 +223,20 @@ bool restoreAttributePeopleData(core::CStateRestoreTraverser& traverser, TSizeUS
         const std::string& name = traverser.name();
         if (name == ATTRIBUTE_TAG) {
             if (core::CStringUtils::stringToType(traverser.value(), lastCid) == false) {
-                LOG_ERROR(<< "Invalid attribute ID in " << traverser.value());
-                return false;
+                LOG_ABORT(<< "Invalid attribute ID in " << traverser.value());
             }
             seenCid = true;
             if (lastCid >= data.size()) {
                 data.resize(lastCid + 1);
             }
         } else if (name == PERSON_TAG) {
-            if (!seenCid) {
-                LOG_ERROR(<< "Incorrect format - person ID before attribute ID in "
+            if (seenCid == false) {
```

Comment:
This should be checking `lastCid < data.size()` as well.

Reply:
> This should be checking `lastCid < data.size()` as well.

I think that this check is already covered by

```
            if (lastCid >= data.size()) {
                data.resize(lastCid + 1);
            }
```
above

Reply:
Yes, good point.